### DISCUSSION TITLE
[Section 3.8 - Modules] Problema Criação Bucket S3

### DISCUSSION THREAD

**Discussion by jacksonpedrini** [chgasparoto/curso-aws-com-terraform#39] [Q&A]
Estou tendo problema para criar a bucket S3.

Segue abaixo o meu código:

```hcl
provider "aws" {
  region = "us-east-1"
}

resource "aws_s3_bucket" "my-tf-test-bucket" {
  bucket = "my-tf-test-bucket18767032"
  acl    = "private"

  tags = {
    Name        = "My bucket"
    Environment = "Dev"
    Managedby = "Terraform"
  }
}
```

E aqui está o erro que está me retornando:

```hcl
aws_s3_bucket.my-tf-test-bucket: Creating...
╷
│ Error: Error creating S3 bucket: AccessDenied: Access Denied
│       status code: 403, request id: EWVW3F96Y4TJWCD5, host id: ihxWaW9XtfnZpkEgJY3hJF84tkvgL/d1r5W17kPBIQuoxGu/eO8gK0qFFmMa7gfsB4L4+7JuOE4=
│
│   with aws_s3_bucket.my-tf-test-bucket,
│   on main.tf line 5, in resource "aws_s3_bucket" "my-tf-test-bucket":
│    5: resource "aws_s3_bucket" "my-tf-test-bucket" {
```

**Comment by chgasparoto** [comment]
@jacksonpedrini você precisa checar as permissões que o usuário que você está usando tem, pois o erro está informando que você não tem acesso para criar um bucket.

O usuário em questão é o que você configurou utilizando o AWS CLI.

**Comment by jacksonpedrini** [comment]
@chgasparoto sim dei ao usuário permissão "AdministratorAccess" conforme pedia a aula e como não funcionou, então mudei para "AmazonS3FullAccess". Porém com ambas as permissões da o erro de "AccessDenied".
Listar a bucket ele lista só não está conseguindo criar, veja abaixo:

```
PS C:\> aws s3api create-bucket --bucket my-bucket1235544 --region us-east-1

An error occurred (AccessDenied) when calling the CreateBucket operation: Access Denied
PS C:\> aws s3 ls
2021-05-26 09:30:44 teste45474712
PS C:\>
```

**Comment by jacksonpedrini** [comment]
@chgasparoto agora deu certo. Encontrei oque estava fazendo de errado. Ao invés de eu setar a permissão do grupo na criação do usuário, acabei setando um limite de permissão. Defini um grupo de permissão e deu tudo certo.. Obrigado pela ajuda.

**Reply by chgasparoto** [reply]
Boa @jacksonpedrini !
